### PR TITLE
Task6

### PR DESCRIPTION
**Pull request by Chmelya**

### CODE REVIEW

File: task6/CacheLibrary/Cache.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace CacheLibrary
+{
+    public class Cache<TKey, TValue>: ICache<TKey, TValue>
+        where TKey : struct
+        where TValue : class
+    {
+        Dictionary<TKey, TValue> cacheVal;
```
Comment (miko96) [inline task6/CacheLibrary/Cache.cs:11]: Can we create  `struct CacheItem` and get rid of cacheTime Dictionary?

File: task6/ConsoleApp/Program.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using CacheLibrary;
+
+namespace ConsoleApp
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Cache<double, string> cache = new Cache<double, string>();
```
Comment (vikariusik) [inline task6/ConsoleApp/Program.cs:10]: the variable cache should be ICache<double, string>

File: task6/CacheLibrary/ICache.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+namespace CacheLibrary
+{
+    interface ICache<TKey, TValue>
```
Comment (vikariusik) [inline task6/CacheLibrary/ICache.cs:5]: Interface should be public

File: task6/CacheLibrary/Cache.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                if (item.Key.Equals(key))
+                {
+                    return item.Time > DateTime.Now ? item.Value : null;
+                }
+            }
+
+            return null;
+        }
+
+        public bool Remove(TKey key)
```
Comment (vikariusik) [inline task6/CacheLibrary/Cache.cs:46]: Try to do this by LINQ

**Pull request merged**